File: app/src/main/java/org/ole/planet/myplanet/base/BaseMemberFragment.java
package org.ole.planet.myplanet.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.ole.planet.myplanet.R;
import org.ole.planet.myplanet.model.RealmUserModel;
import org.ole.planet.myplanet.ui.team.BaseTeamFragment;

import java.util.List;

public abstract class BaseMemberFragment extends BaseTeamFragment {
    public abstract List<RealmUserModel> getList();
    public abstract RecyclerView.Adapter getAdapter();
    public abstract RecyclerView.LayoutManager getLayoutManager();
    public RecyclerView rvMember;
    public TextView tvNodata;




    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_members, container, false);
        rvMember = v.findViewById(R.id.rv_member);
        tvNodata = v.findViewById(R.id.tv_nodata);
        return v;
    }




    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        rvMember.setLayoutManager(getLayoutManager());
        rvMember.setAdapter(getAdapter());
        showNoData(tvNodata, getList().size());
    }
}

File: app/src/main/java/org/ole/planet/myplanet/base/BaseMemberFragment.kt
package org.ole.planet.myplanet.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import org.ole.planet.myplanet.databinding.FragmentMembersBinding
import org.ole.planet.myplanet.model.RealmUserModel
import org.ole.planet.myplanet.ui.teams.BaseTeamFragment

abstract class BaseMemberFragment : BaseTeamFragment() {
    abstract val list: List<RealmUserModel?>
    abstract val adapter: RecyclerView.Adapter<*>?
    abstract val layoutManager: RecyclerView.LayoutManager?
    private var _binding: FragmentMembersBinding? = null
    protected val binding get() = _binding!!
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentMembersBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.rvMember.layoutManager = layoutManager
        binding.rvMember.adapter = adapter
        showNoData(binding.tvNodata, list.size, "members")
    }

    override fun onDestroyView() {
        _binding = null
        super.onDestroyView()
    }
}
